### PR TITLE
fix(condo): DOMA-9746 deeplink to mini-app after registration

### PR DESCRIPTION
**Pull request by tolmachev21**

### CODE REVIEW

File: apps/condo/domains/organization/components/OrganizationSelect.tsx [hunk 1]
```diff
@@ -78,8 +80,12 @@ export const InlineOrganizationSelect: React.FC = () => {
     const { setIsVisible: showCreateOrganizationModal, ModalForm: CreateOrganizationModalForm } = useCreateOrganizationModalForm({
         onFinish: async (createdOrganization) => {
             const organizationType = get(createdOrganization, 'type')
+            const isValidNextUrl = next && !Array.isArray(next) && isSafeUrl(next)
```
Comment (toplenboren) [inline apps/condo/domains/organization/components/OrganizationSelect.tsx:83]: doesnt isSafeUrl check `next` type already?

File: apps/condo/domains/organization/components/OrganizationSelect.tsx [hunk 2]
```diff
@@ -78,8 +80,12 @@ export const InlineOrganizationSelect: React.FC = () => {
...
 
-            if (organizationType === MANAGING_COMPANY_TYPE) {
+            // After registration and created first organization we want redirect user to nextUrl
```
Comment (toplenboren) [inline apps/condo/domains/organization/components/OrganizationSelect.tsx:85]:
```suggestion
            // Redirects user to `next` url after signup
```

File: apps/condo/domains/organization/components/OrganizationSelect.tsx [hunk 3]
```diff
@@ -78,8 +80,12 @@ export const InlineOrganizationSelect: React.FC = () => {
...
+            if (isValidNextUrl && !hasInvites && !link && !filteredEmployees.length) {
+                await router.push(next)
+            } else if (organizationType === MANAGING_COMPANY_TYPE) {
```
Comment (toplenboren) [inline apps/condo/domains/organization/components/OrganizationSelect.tsx:88]: Why wouldn't it work for management companies?

File: apps/condo/domains/organization/components/OrganizationSelect.tsx [hunk 4]
```diff
@@ -78,8 +80,12 @@ export const InlineOrganizationSelect: React.FC = () => {
     const { setIsVisible: showCreateOrganizationModal, ModalForm: CreateOrganizationModalForm } = useCreateOrganizationModalForm({
         onFinish: async (createdOrganization) => {
             const organizationType = get(createdOrganization, 'type')
+            const isValidNextUrl = next && !Array.isArray(next) && isSafeUrl(next)
 
-            if (organizationType === MANAGING_COMPANY_TYPE) {
+            // After registration and created first organization we want redirect user to nextUrl
+            if (isValidNextUrl && !hasInvites && !link && !filteredEmployees.length) {
```
Comment (toplenboren) [inline apps/condo/domains/organization/components/OrganizationSelect.tsx:86]: - why !hasInvites blocks redirect to `next` ?

- why `!link` blocks redirect?

Comment (tolmachev21) [reply apps/condo/domains/organization/components/OrganizationSelect.tsx:85]: We need to redirect to nextUrl only after registration and first creation of the organization. Therefore there is a check for lint, filteredEmployees, hasInvites, isValidNextUrl

File: apps/condo/pages/auth/register.tsx [hunk 5]
```diff
@@ -31,12 +31,17 @@ const RegisterPage: AuthPage = () => {
...
     const router = useRouter()
+    const { query: { next }  } = router
 
     const { token, isConfirmed, tokenError, setToken, setTokenError } = useContext(RegisterContext)
     const [step, setStep] = useState('inputPhone')
 
     const handleFinish = useCallback(async () => {
-        await router.push('/')
+        if (next) {
+            await router.push(`/?next=${next}`)
```
Comment (SavelevMatthew) [inline apps/condo/pages/auth/register.tsx:41]: Strange, next must only available on auth pages

**Review by SavelevMatthew — CHANGES_REQUESTED** [review]
Seems strange to me, why all pages do accept `next` parameter now?

File: apps/condo/domains/organization/components/OrganizationSelect.tsx
[the remark this answers is not in the document]
Comment (tolmachev21) [reply apps/condo/domains/organization/components/OrganizationSelect.tsx:1]: I am understand this, but after registration and creation of the organization, although we get a page with /nextUrl, however after a few seconds we are redireced on /tour. So we need to change the code in the index. How I can change this behaviore?

File: apps/condo/domains/organization/components/OrganizationSelect.tsx (hunk 4 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/organization/components/OrganizationSelect.tsx:85]: If user has invites/employees, then he is not be redirected to `next` link.
Are you sure that it is correct logic?
I think if the `next` parameter is specified, then the user should be redirected there.

**Review by SavelevMatthew — APPROVED** [review]

**Pull request merged**